src/main/java/bio/terra/service/filedata/azure/tables/TableDirectoryDao.java

```diff
@@ -199,6 +201,7 @@ public FireStoreDirectoryEntry retrieveByPath(
     TableClient tableClient = tableServiceClient.getTableClient(TABLE_NAME);
     ListEntitiesOptions options =
         new ListEntitiesOptions().setFilter(String.format("path eq '%s'", dirPath));
+    // TODO - add check that there are entries
```

Comment:
With this TODO comment, I think I was thinking of adding the check to make sure the table exists/has entities.  But, not sure if it's really needed - totally up to you.
Something like:
`TableServiceClientUtils.tableHasEntries(tableServiceClient, TABLE_NAME)`